# Data Analysis Project: 5 - Pet Shelter Analysis

In [1]:
import pandas as pd
intakes_outcomes = pd.read_csv('./archive/aac_intakes_outcomes.csv')
intakes = pd.read_csv('./archive/aac_intakes.csv')
outcomes = pd.read_csv('./archive/aac_outcomes.csv')

In [2]:
intakes_outcomes.head(3)

age_upon_outcome animal_id_outcome        date_of_birth outcome_subtype   
0         10 years           A006100  2007-07-09 00:00:00             NaN  \
1          7 years           A006100  2007-07-09 00:00:00             NaN   
2          6 years           A006100  2007-07-09 00:00:00             NaN   

      outcome_type sex_upon_outcome  age_upon_outcome_(days)   
0  Return to Owner    Neutered Male                     3650  \
1  Return to Owner    Neutered Male                     2555   
2  Return to Owner    Neutered Male                     2190   

   age_upon_outcome_(years) age_upon_outcome_age_group     outcome_datetime   
0                      10.0                (7.5, 10.0]  2017-12-07 14:07:00  \
1                       7.0                 (5.0, 7.5]  2014-12-20 16:35:00   
2                       6.0                 (5.0, 7.5]  2014-03-08 17:10:00   

   ...  age_upon_intake_age_group      intake_datetime intake_month   
0  ...                (7.5, 10.0]  2017-12-07 00:00:00           12  \
1  ...                 (5.0, 7.5]  2014-12-19 10:21:00           12   
2  ...                 (5.0, 7.5]  2014-03-07 14:26:00            3   

  intake_year  intake_monthyear  intake_weekday  intake_hour  intake_number   
0        2017           2017-12        Thursday           14            1.0  \
1        2014           2014-12          Friday           10            2.0   
2        2014           2014-03          Friday           14            3.0   

             time_in_shelter time_in_shelter_days  
0  0 days 14:07:00.000000000             0.588194  
1  1 days 06:14:00.000000000             1.259722  
2  1 days 02:44:00.000000000             1.113889  

[3 rows x 41 columns]

In [3]:
intakes.head(3)

age_upon_intake animal_id animal_type                     breed   
0         8 years   A706918         Dog  English Springer Spaniel  \
1       11 months   A724273         Dog               Basenji Mix   
2         4 weeks   A665644         Cat    Domestic Shorthair Mix   

         color                 datetime                datetime2   
0  White/Liver  2015-07-05T12:59:00.000  2015-07-05T12:59:00.000  \
1  Sable/White  2016-04-14T18:43:00.000  2016-04-14T18:43:00.000   
2       Calico  2013-10-21T07:59:00.000  2013-10-21T07:59:00.000   

                       found_location intake_condition intake_type     name   
0    9409 Bluegrass Dr in Austin (TX)           Normal       Stray    Belle  \
1  2818 Palomino Trail in Austin (TX)           Normal       Stray  Runster   
2                         Austin (TX)             Sick       Stray      NaN   

  sex_upon_intake  
0   Spayed Female  
1     Intact Male  
2   Intact Female

In [4]:
outcomes.head(3)

age_upon_outcome animal_id animal_type                   breed   
0          2 weeks   A684346         Cat  Domestic Shorthair Mix  \
1           1 year   A666430         Dog              Beagle Mix   
2           1 year   A675708         Dog                Pit Bull   

          color        date_of_birth             datetime   
0  Orange Tabby  2014-07-07T00:00:00  2014-07-22T16:04:00  \
1   White/Brown  2012-11-06T00:00:00  2013-11-07T11:47:00   
2    Blue/White  2013-03-31T00:00:00  2014-06-03T14:20:00   

             monthyear     name outcome_subtype outcome_type sex_upon_outcome  
0  2014-07-22T16:04:00      NaN         Partner     Transfer      Intact Male  
1  2013-11-07T11:47:00     Lucy         Partner     Transfer    Spayed Female  
2  2014-06-03T14:20:00  *Johnny             NaN     Adoption    Neutered Male

## Meets Expectations Requirements:

### 1. Is there an area where more pets are found?
- Find the top 5 places where animals are found so the shelter can coordinate with local volunteers and animal control to monitor these areas.

In [24]:
# count how many times the same values repeat in this "found_location" series
locations = intakes['found_location'].value_counts()[2:7]
locations

found_location
Travis (TX)                          921
7201 Levander Loop in Austin (TX)    517
Del Valle (TX)                       411
Pflugerville (TX)                    372
Manor (TX)                           280
Name: count, dtype: int64

#### Answer: 
    - 1: Travis :)
    - 2: 7201 Levander Loop in Austin
    - 3: Del Valle
    - 4: Pflugerville
    - 5: Manor

#### Solution:
To gather this data I utilized the `value_counts` method on the `'found_location'` series and sliced the first two entries away because they were simply `Austin (TX)`, which is where the shelter is located, and `Outside Jurisdiction` where I assume local volunteers and animal control would not be allowed to monitor.

### 2-a. What is the average number of pets found in a month in the year 2015?

### 2-b. Are there months where there is a higher number of animals found?

### 3. What is the ratio of incoming pets vs. adopted pets?

### 4. What is the distribution of the types of animals in the shelter?

### 5. What are the adoption rates for specific breeds?

### 6. What are the adoption rates for different colorings?

### 7. About how many animals are spayed/neutered each month?

## Exceeds Expectations Requirements:

### 1-a. How many animals in the shelter are repeats?

### 1-b. Which animal was returned to the shelter the most?

### 2. What are the adoption rates for the following age groups?
- baby: 4 months and less
- young: 5 months - 2 years
- adult: 3 years - 10 years
- senior: 11+

### 3. If spay/neuter for a dog costs $100 and a spay/neuter for a cat costs $50, how much did the shelter spend in 2015 on these procedures?